src/Atmos/Model/moisture.jl

```diff
@@ -0,0 +1,95 @@
+#### Moisture component in atmosphere model
+abstract type MoistureModel end
+
+vars_state(::MoistureModel, T) = Tuple{}
+vars_transform(::MoistureModel, T) = Tuple{}
+vars_diffusive(::MoistureModel, T) = Tuple{}
+vars_aux(::MoistureModel, T) = Tuple{}
+
+function diffusive!(::MoistureModel, diffusive, ∇transform, state, aux, t, ν)
+end
+function flux_diffusive!(::MoistureModel, flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+end
+function update_aux!(::MoistureModel, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+end
+function gradtransform!(::MoistureModel, transform::Vars, state::Vars, aux::Vars, t::Real)
+end
+
+
+
+struct EquilMoist <: MoistureModel
+end
+vars_state(::EquilMoist,T) = NamedTuple{(:ρq_tot,),Tuple{T}}
+vars_transform(::EquilMoist,T) = NamedTuple{(:q_tot),Tuple{T}}
+vars_diffusive(::EquilMoist,T) = NamedTuple{(:ρd_q_tot, :ρJ_ρD), Tuple{SVector{3,T},SVector{3,T}}}
+vars_aux(::EquilMoist,T) = NamedTuple{(:e_int, :temperature),Tuple{T,T}}
+
+
+function pressure(::EquilMoist, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  q_tot = stat.ρq_tot / ρ
+  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  air_pressure(phase)
+end
+function soundspeed(::EquilMoist, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  q_tot = stat.ρq_tot / ρ
+  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
```

Comment:
Is there a simple way to generalize this is condensate (`q_liq` and `q_ice`) are part of the state? @charleskawczynski ?

Reply:
Yes: I think the idea is that we can have different moisture submodels: this one calculates them based on thermodynamic equilibrium, others could treat them as state variables.